### PR TITLE
UI changes for Attachment widget

### LINKED ISSUE #1796: Update Attachment widget UI to be more consistent
**Is your feature request related to a problem? Please describe.**
@jingtang10 has some concern that the current Attachment widget design won't be consistent for single and multi attachment.

**Describe the solution you'd like**
@shelaghm provided a new design in which it will be consistent for single and multi attachment.

**Describe alternatives you've considered**
N/A

**Additional context**

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by FikriMilano**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1796

**Description**
- [Create independent file preview UI](https://github.com/google/android-fhir/pull/1811/commits/0f006087e009c308138cec11f28bb6467847d012)
- [Create independent photo preview UI](https://github.com/google/android-fhir/pull/1811/commits/930b0bd39fded6fbec051e59b09e1a14d88e767d)
- [Update attachment widget UI](https://github.com/google/android-fhir/pull/1811/commits/863979f2875d9672713a6241155620f603d00106)
- [Refactor Attachment widget logic](https://github.com/google/android-fhir/pull/1811/commits/cbc0cd8711da670a9fe12ce0fa5fbdc2ac37af50) 
  - Update the previous delete button into photoDeleteButton and fileDeleteButton,
same functionality but different xml file
  - Add line divider visibility logic
  - Add labelUploaded visibility logic
  - Remove methods to display or hide delete button and instead
moved the visibility logic to displayPreview method
  - Rename variables
  - Re-arrange code lines, looks neater

**Alternative(s) considered**
N/A

**Type**
Choose one: Feature

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by FikriMilano** [comment]
Kindly review this PR after https://github.com/google/android-fhir/pull/1680 is merged

**Comment by omarismail94** [comment]
@FikriMilano can you resolve the branch conflicts ?

**Comment by FikriMilano** [comment]
@omarismail94 resolved

**Comment by omarismail94** [comment]
@FikriMilano sorry, one more thing. Can you add screenshots of the UI changes that this PR introduces? We can then get a UX thumbs up from @shelaghm

**Comment by FikriMilano** [comment]
@shelaghm @omarismail94 
screenshots incomingg, your thoughts please

**Comment by FikriMilano** [comment]
@shelaghm sure, that would be no problem

**Comment by FikriMilano** [comment]
@omarismail94 resolved

**Comment by shelaghm** [comment]
@FikriMilano thank you!

**Comment by omarismail94** [comment]
thank you @shelaghm and @FikriMilano ! I will have time to review this today

**Review by omarismail94 — COMMENTED** [review]
@FikriMilano the changes look good, Im just curious, how are we limiting image or file upload sizes? Is this happening in this PR, or is that in a follow up PR?

**Comment by FikriMilano** [comment]
@omarismail94 that has been implemented previously on PR https://github.com/google/android-fhir/pull/1680

You can also have a look at this code about how we limit upload file size https://github.com/google/android-fhir/blob/master/datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemAttachmentViewHolderFactory.kt#L232-L239

**Comment by jingtang10** [comment]
can you please update the description of the PR to have a list of what changes are included?

**Comment by jingtang10** [comment]
I think it would be nice to think about how this would help pave the way for repeated attahcments. I think the new layout files you added could be reused to populate new previews, right?

**Comment by FikriMilano** [comment]
@jingtang10 yeah, that's right.

also, are you implying we use some sort of a layout to contain the previews? and the previews will be the child view of that layout. So, instead updating the visibility, we add or remove preview from a LinearLayout.

File: datacapture/src/main/res/drawable/rounded_corner_preview_filled.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<shape
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle"
+>
+    <solid android:color="#D4E3FD" />
```
Comment (jingtang10) [inline datacapture/src/main/res/drawable/rounded_corner_preview_filled.xml:6]: this is a hardcoded colour that cannot be customized by developers?

**Review by jingtang10 — APPROVED** [review]
Thanks for this this looks good to me! Well done!

please wait to see if @omarismail94 has any additional comments before merging.

File: datacapture/src/main/res/drawable/rounded_corner_preview_filled.xml (hunk 1 above)
Comment (FikriMilano) [reply datacapture/src/main/res/drawable/rounded_corner_preview_filled.xml:6]: will change these to color attributes

**Review by omarismail94 — APPROVED** [review]
Great change Fikri!!

**Pull request merged**